Clarify eglot behavior

non-Jedi opened the pull request:
Eglot treats the offset as characters by default (codepoints).
https://github.com/joaotavora/eglot/pull/125

Eglot and lsp-mode no longer listed as "work in progress" on https://langserver.org

non-Jedi committed:
Clarify eglot behavior

Eglot treats the offset as characters by default (codepoints).
https://github.com/joaotavora/eglot/pull/125

Eglot and lsp-mode no longer listed as "work in progress" on https://langserver.org

non-Jedi committed:
Update unit used by sourcegraph emacs-lsp

It uses `(current-column)` for line position which is only related to
how emacs visually displays the file. Most of the time this is
equivalent to the unit being grapheme clusters though there will of
course be exceptions based on emacs font rendering.

Avi-D-coder committed:
Update README.md

Avi-D-coder commented:
Where exactly does it say eglot uses codepoints?

non-Jedi commented:
In the issue they use the word "character" rather than codepoint. When looking at the code, eglot uses [`(- (point) (point-at-bol))`](https://github.com/joaotavora/eglot/blob/74240c7dae22716c6925341d0e76fdc76f37190f/eglot.el#L978) by default. [The documentation around "point"](https://www.gnu.org/software/emacs/manual/html_node/elisp/Point.html) just uses the word "character" again. But [elsewhere in the manual](https://www.gnu.org/software/emacs/manual/html_node/elisp/Text-Representations.html#Text-Representations), they say that each character corresponds to a unicode codepoint.

non-Jedi commented:
By the way, I couldn't make the "client counts" add up the way you showed. You may want to recount.

Avi-D-coder commented:
> By the way, I couldn't make the "client counts" add up the way you showed. You may want to recount.

That would be great thanks.

Keep in mind "Multiple implementations in the same repo or derived from a shared dependency are counted once." While technically all the JS/TS implementation use the same Microsoft libraries, Non Microsoft JS implementations can be counted individually.

non-Jedi committed:
Update client counts

Did not count https://github.com/TypeFox/monaco-languageclient since
it depends on a vscode npm module
